src/CorePolicyTest.cc

```diff
@@ -0,0 +1,179 @@
+    }
+
+    virtual void TearDown()
+    {
+        // Unblock all cores so they can shut down and be joined.
+        coreArbiter->setRequestedCores({Arachne::maxNumCores,0,0,0,0,0,0,0});
+
+        shutDown();
+        waitForTermination();
+    }
+};
+
+// Helper function for tests with timing dependencies, so that we wait for a
+// finite amount of time in the case of a bug causing an infinite loop.
+static void limitedTimeWait(std::function<bool()> condition,
+                int numIterations) {
+    for (int i = 0; i < numIterations; i++) {
+        if (condition()) {
+            break;
+        }
+        usleep(1000);
+    }
+    // We use assert here because an infinite loop will result in TearDown
+    // not being able to complete, so we might as well terminate the tests here.
+    ASSERT_TRUE(condition());
+}
+
+TEST_F(ArachneTest, CorePolicy_constructor) {
+    CorePolicy* corePolicy = new CorePolicy();
+    for (unsigned i = 0; i < std::thread::hardware_concurrency(); i++) {
+        EXPECT_EQ(corePolicy->threadCoreMap[corePolicy->baseClass]->map[i], 0);
+    }
+    delete corePolicy;
+}
+
+TEST_F(ArachneTest, CorePolicy_addCore) {
+    CorePolicy* corePolicy = new CorePolicy();
+    corePolicy->addCore(5);
+    EXPECT_EQ(corePolicy->threadCoreMap[corePolicy->baseClass]->map[0], 5);
+    corePolicy->addCore(4);
+    corePolicy->addCore(7);
+    EXPECT_EQ(corePolicy->threadCoreMap[corePolicy->baseClass]->map[0], 5);
+    EXPECT_EQ(corePolicy->threadCoreMap[corePolicy->baseClass]->map[1], 4);
+    EXPECT_EQ(corePolicy->threadCoreMap[corePolicy->baseClass]->map[2], 7);
+    EXPECT_EQ(corePolicy->threadCoreMap[corePolicy->baseClass]->numFilled, 3U);
+    delete corePolicy;
+}
+
+TEST_F(ArachneTest, CorePolicy_removeCore) {
+    CorePolicy* corePolicy = new CorePolicy();
+    corePolicy->addCore(5);
+    corePolicy->addCore(4);
+    corePolicy->addCore(7);
+    corePolicy->removeCore(5);
+    EXPECT_EQ(corePolicy->threadCoreMap[corePolicy->baseClass]->numFilled, 2U);
+    EXPECT_EQ(corePolicy->threadCoreMap[corePolicy->baseClass]->map[0], 7);
+    EXPECT_EQ(corePolicy->threadCoreMap[corePolicy->baseClass]->map[1], 4);
+    corePolicy->removeCore(4);
+    EXPECT_EQ(corePolicy->threadCoreMap[corePolicy->baseClass]->numFilled, 1U);
+    EXPECT_EQ(corePolicy->threadCoreMap[corePolicy->baseClass]->map[0], 7);
+    corePolicy->removeCore(7);
+    EXPECT_EQ(corePolicy->threadCoreMap[corePolicy->baseClass]->numFilled, 0U);
+    delete corePolicy;
+}
+
+TEST_F(ArachneTest, CorePolicy_getThreadCoreMapEntry) {
+    CorePolicy* corePolicy = new CorePolicy();
+    threadCoreMapEntry* entry = corePolicy->getThreadCoreMapEntry(corePolicy->baseClass);
+    corePolicy->addCore(5);
+    EXPECT_EQ(entry->map[0], 5);
+    corePolicy->addCore(4);
+    corePolicy->addCore(7);
+    EXPECT_EQ(entry->map[0], 5);
+    EXPECT_EQ(entry->map[1], 4);
+    EXPECT_EQ(entry->map[2], 7);
+    EXPECT_EQ(entry->numFilled, 3U);
+    delete corePolicy;
+}
```

Comment:
This is a decent start on tests, but it doesn't test several methods at all, such as coreLoadEstimator. Let's discuss.